pyciemss/interfaces.py

```diff
@@ -785,11 +788,11 @@ def optimize(
             - A callable function defining the quantity of interest to optimize over.
         risk_bounds: float
             - The threshold on the risk constraint.
-        static_parameter_interventions: Dict[torch.Tensor, str]
-            - A dictionary of the form {intervention_time: parameter_name}
-              of static parameter interventions to optimize over.
-            - Each key is the time at which the intervention is applied.
-            - Each value is a string with the intervention parameter name.
+        static_parameter_interventions: Callable[[torch.Tensor], Dict[float, Dict[str, Intervention]]]
```

Comment:
Could you move this into a separate PR. A good fix, but unrelated to adding warnings.

Reply:
I have now added this to #544. You can merge that first and then this PR.